pkg/resources/object_renaming_acceptance_test.go

```diff
@@ -0,0 +1,1366 @@
+		ProtoV6ProviderFactories: acc.TestAccProtoV6ProviderFactories,
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.RequireAbove(tfversion.Version1_5_0),
+		},
+		CheckDestroy: acc.CheckDestroy(t, resources.Schema),
+		Steps: []resource.TestStep{
+			{
+				Config: config.FromModels(t, schemaModelConfig, databaseConfigModel),
+			},
+			{
+				PreConfig: func() {
+					newDatabaseId := acc.TestClient().Ids.RandomAccountObjectIdentifier()
+					acc.TestClient().Database.Alter(t, databaseId, &sdk.AlterDatabaseOptions{NewName: &newDatabaseId})
+					t.Cleanup(acc.TestClient().Database.DropDatabaseFunc(t, newDatabaseId))
+				},
+				ConfigPlanChecks: resource.ConfigPlanChecks{
+					PreApply: []plancheck.PlanCheck{
+						plancheck.ExpectResourceAction("snowflake_database.test", plancheck.ResourceActionCreate),
+						plancheck.ExpectResourceAction("snowflake_schema.test", plancheck.ResourceActionCreate),
+					},
+				},
+				Config:      config.FromModels(t, schemaModelConfig, databaseConfigModel),
+				ExpectError: regexp.MustCompile("does not exist or not authorized"),
+			},
+		},
+	})
+}
+
+func TestAcc_ShallowHierarchy_IsInConfig_RenamedExternally_WithoutDependency_AfterRenameDatabaseReferencingOldNameAndSchemaReferencingNewDatabaseName(t *testing.T) {
+	_ = testenvs.GetOrSkipTest(t, testenvs.EnableObjectRenamingTest)
+	_ = testenvs.GetOrSkipTest(t, testenvs.EnableAcceptance)
+	acc.TestAccPreCheck(t)
+
+	databaseId := acc.TestClient().Ids.RandomAccountObjectIdentifier()
+	newDatabaseId := acc.TestClient().Ids.RandomAccountObjectIdentifier()
+	schemaName := acc.TestClient().Ids.Alpha()
+
+	databaseConfigModel := model.Database("test", databaseId.Name())
+	schemaModelConfig := model.Schema("test", databaseId.Name(), schemaName)
+	schemaModelConfigWithNewDatabaseId := model.Schema("test", newDatabaseId.Name(), schemaName)
+
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: acc.TestAccProtoV6ProviderFactories,
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.RequireAbove(tfversion.Version1_5_0),
+		},
+		CheckDestroy: acc.CheckDestroy(t, resources.Schema),
+		Steps: []resource.TestStep{
+			{
+				Config: config.FromModels(t, databaseConfigModel, schemaModelConfig),
+			},
+			{
+				PreConfig: func() {
+					acc.TestClient().Database.Alter(t, databaseId, &sdk.AlterDatabaseOptions{NewName: &newDatabaseId})
+					t.Cleanup(acc.TestClient().Database.DropDatabaseFunc(t, newDatabaseId))
+				},
+				ConfigPlanChecks: resource.ConfigPlanChecks{
+					PreApply: []plancheck.PlanCheck{
+						plancheck.ExpectResourceAction("snowflake_database.test", plancheck.ResourceActionCreate),
+						plancheck.ExpectResourceAction("snowflake_schema.test", plancheck.ResourceActionCreate),
+					},
+				},
+				Config:      config.FromModels(t, databaseConfigModel, schemaModelConfigWithNewDatabaseId),
+				ExpectError: regexp.MustCompile("Failed to create schema"), // already exists
```

Comment:
Not really, from the test title `_AfterRenameDatabaseReferencingOldNameAndSchemaReferencingNewDatabaseName` schema is referencing a new database name. As we talked about it, changing database of schema can make it remove itself from state in Read which is good, then the terraform tries to create this schema, but fails because schema already exists in new database. Don't know if that answers the question, I'll write additional comment to help with explanation.